contracts/RestrictedNORI.sol

```diff
@@ -947,31 +947,29 @@ contract RestrictedNORI is
    * Linearly released balance for a single schedule at the current block timestamp, ignoring any
    * released amount floor that has been set for the schedule.
    */
-  /* solhint-disable not-rely-on-time, this is time-dependent */
   function _linearReleaseAmountAvailable(uint256 scheduleId)
     internal
     view
     returns (uint256)
   {
+    /* solhint-disable not-rely-on-time, this is time-dependent */
     Schedule storage schedule = _scheduleIdToScheduleStruct[scheduleId];
     uint256 linearAmountAvailable;
     // slither-disable-next-line timestamp this contract is block time dependent
-
     if (block.timestamp >= schedule.endTime) {
       linearAmountAvailable = totalSupply(scheduleId);
     } else {
       uint256 rampTotalTime = schedule.endTime - schedule.startTime;
       // slither-disable-next-line timestamp this contract is block time dependent
       linearAmountAvailable = block.timestamp < schedule.startTime
         ? 0
-        : (_scheduleTrueTotal(schedule, scheduleId) *
+        : (_scheduleTrueTotal(schedule.totalClaimedAmount, scheduleId) *
```

Comment:
function internally reloads the schedule from storage resulting in  `Schedule storage schedule = _scheduleIdToScheduleStruct[scheduleId];` occurring n*2 times

Reply:
internally it only calls `totalSupply` ... might be looking at old commit?  I may refactor this so that the total supply is passed

Reply:
Sorry, maybe commented on the wrong line. I think I was thinking about [this](https://github.com/nori-dot-eco/contracts/blob/f79c538a69dec1b88e8eed82ce5e1823c5e7bac2/contracts/RestrictedNORI.sol#L431)